doc/source/whatsnew/v0.17.0.txt

```diff
@@ -629,4 +629,5 @@ Bug Fixes
 - Bug in ``pd.DataFrame`` when constructing an empty DataFrame with a string dtype (:issue:`9428`)
 - Bug in ``pd.unique`` for arrays with the ``datetime64`` or ``timedelta64`` dtype that meant an array with object dtype was returned instead the original dtype (:issue: `9431`)
 - Bug in ``DatetimeIndex.take`` and ``TimedeltaIndex.take`` may not raise ``IndexError`` against invalid index (:issue:`10295`)
+- Bug in ``tslib.cast_to_nanoseconds`` with ``NaT`` (:issue:`10747`) 
```

Comment:
say something more useful to a reader, e.g. ``Series([np.nan]).asytype('M8[ms]')`` now correctly returns ``Series([pd.NaT]')``